tests/core/test_core_containers.cpp

```diff
@@ -71,6 +71,48 @@ TEST_F(core_container_test, device_vector) {
     }
 }
 
+/// Test(s) for @c vecmem::device_vector
+TEST_F(core_container_test, device_nonconst_vector) {
+
+    vecmem::device_vector<int> test_vector(
+        vecmem::get_data(m_reference_vector));
+    EXPECT_EQ(test_vector.size(), m_reference_vector.size());
+    EXPECT_EQ(test_vector.empty(), m_reference_vector.empty());
+    EXPECT_TRUE(std::equal(m_reference_vector.begin(), m_reference_vector.end(),
+                           test_vector.begin()));
+    EXPECT_EQ(std::accumulate(test_vector.begin(), test_vector.end(), 0),
+              std::accumulate(test_vector.rbegin(), test_vector.rend(), 0));
+
+    decltype(test_vector)::const_iterator itfc1 = test_vector.cbegin();
+    decltype(test_vector)::const_iterator itfc2 = test_vector.cend();
+    decltype(test_vector)::iterator itfn1 = test_vector.begin();
+    decltype(test_vector)::iterator itfn2 = test_vector.end();
+    decltype(test_vector)::const_reverse_iterator itrc1 = test_vector.crbegin();
+    decltype(test_vector)::const_reverse_iterator itrc2 = test_vector.crend();
+    decltype(test_vector)::reverse_iterator itrn1 = test_vector.rbegin();
+    decltype(test_vector)::reverse_iterator itrn2 = test_vector.rend();
+
+    EXPECT_TRUE(itfn1 == itfc1);
+    EXPECT_TRUE(itfc1 == itfn1);
+    EXPECT_TRUE(itfn2 != itfc1);
+    EXPECT_TRUE(itfc1 != itfn2);
+    EXPECT_TRUE(itrn1 == itrc1);
+    EXPECT_TRUE(itrc1 == itrn1);
+    EXPECT_TRUE(itrn2 != itrc1);
+    EXPECT_TRUE(itrc1 != itrn2);
+    EXPECT_TRUE(itfc2 == itfn2);
+    EXPECT_TRUE(itfc2 != itfn1);
+    EXPECT_TRUE(itrc2 == itrn2);
+    EXPECT_TRUE(itrc2 != itrn1);
```

Comment:
Let's rather use `EXPECT_EQ(...)` / `EXPECT_NEQ(...)` on these. I believe those should use the comparison operators correctly. 🤔

Reply:
Sure, I like being explicit here as we're specifically testing the `==` and `!=` operators but this probably works, too.